code/go/internal/validator/semantic/validate_docs_structure.go

```diff
@@ -0,0 +1,242 @@
+			),
+		}
+	}
+
+	err = validateReadmeStructure(fsys, enforcedSections)
+
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				err,
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+	return nil
+}
+
+func shouldValidateDocsStructure(fsys fspath.FS) (*specerrors.DocsStructureEnforced, error) {
+	validationPath := "validation.yml"
+	files, err := pkgpath.Files(fsys, validationPath)
+	if err != nil || len(files) == 0 {
+		return nil, nil
+	}
+
+	validationFile := files[0]
+	data, err := validationFile.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %s: %w", validationFile.Path(), err)
+	}
+
+	var config specerrors.ConfigFilter
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal YAML from %s: %w", validationFile.Path(), err)
+	}
+
+	if config.DocsStructureEnforced.Enabled {
+		return &config.DocsStructureEnforced, nil
+	}
+
+	return nil, nil
+}
+
+func readDocsStructureEnforcedConfig(fsys fspath.FS, config *specerrors.DocsStructureEnforced) ([]string, error) {
+	defaultSections, err := loadSectionsFromConfig(fmt.Sprintf("%d", config.Version))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load enforced sections from config: %w", err)
+	}
+
+	var sections []string
+	for _, section := range defaultSections {
+		if contains(config.DocsStructureSkip, section) {
+			continue
+		}
+		sections = append(sections, section)
+	}
+
+	return sections, nil
+}
+
+func contains(slice []specerrors.DocsStructureSkip, item string) bool {
+	for _, s := range slice {
+		if s.Title == item {
+			return true
+		}
+	}
+	return false
+}
+
+func loadSectionsFromConfig(version string) ([]string, error) {
+	var schemaPath string
+	switch version {
+	case "1":
+		schemaPath = "enforced_sections_v1.yml"
+	default:
+		return nil, fmt.Errorf("unsupported version: %s", version)
+	}
+
+	data, err := fs.ReadFile(spec.DocsFS(), schemaPath)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to read schema file: %w", err)
+	}
+
+	var spec EnforcedSections
+	if err := yaml.Unmarshal(data, &spec); err != nil {
+		return nil, fmt.Errorf("invalid schema YAML: %w", err)
+	}
+
+	if spec.Version != version {
+		return nil, fmt.Errorf("schema version mismatch: got %s, expected %s", spec.Version, version)
+	}
+
+	sections := make([]string, 0, len(spec.Sections))
+	for _, section := range spec.Sections {
+		if section.Name != "" {
+			sections = append(sections, section.Name)
+		}
+	}
+
+	return sections, nil
+}
+
+func validateReadmeStructure(fsys fspath.FS, enforcedSections []string) error {
+	var errs DocsValidationError
+	files, err := pkgpath.Files(fsys, "docs/*.md")
+	if err != nil {
+		return fmt.Errorf("docs folder %s not found: %w", "docs/*.md", err)
+	}
+	if len(files) == 0 {
+		return nil
+	}
+
+	for _, file := range files {
+		if !file.IsDir() {
+			fullPath := path.Join("docs", file.Name())
+			content, err := fs.ReadFile(fsys, fullPath)
```

Comment:
Could the new helper be used here?
```suggestion
			content, err := file.ReadAll()
```